### DISCUSSION TITLE
What's the meaning of "xc" in clut paramter?

### DISCUSSION THREAD

**Discussion by keyurshah070** [ImageMagick/ImageMagick#2307] [Help] [answered]
I am converting the Image with a different color in PHP Language so I have below a command which works,

`magick \( lorem.jpg -colorspace gray -contrast-stretch 0 \) \( -size 1x1 xc:"#f48023" xc:white +append -filter triangle -resize 1024 \) -clut lorem_ipsum_recolor2.png
`

but I am not getting the meaning of the "xc" here , can someone explain me this command?

**Comment by dlemstra** [comment, marked as answer]
You are reading a file with a specific color. The `xc:` reader reads an image that with a constant color. In this case you are reading an image with the color `#f48023` with a 1x1 dimension (`-size 1x1`).

**Reply by keyurshah070** [reply]
but my Image does not contain the color "#f48023" , it is converting the any Image with "#f48023" color, then what's the meaning of "white" here?

**Reply by dlemstra** [reply]
ImageMagick allows you to read multiple images. You are reading an orange (`#f48023`) and a white (`xc:white`) image and with `+append` you are combing images.

**Reply by keyurshah070** [reply]
apologies for maybe silly question , (As I am new to ImageMagick) so in the command,

magick \( lorem.jpg -colorspace gray -contrast-stretch 0 \) \( -size 1x1 xc:"#f48023" xc:white +append -filter triangle -resize 1024 \) -clut lorem_ipsum_recolor2.png

- It creates the Image 1*1 with color #f48023

- It creates the Image 1*1 with color white

- then combine both Images

- then it cult the Image?

- what about -colorspace gray -contrast-stretch 0 ?

thank you so much in advance for the answer

**Reply by fmw42** [reply]
You are reading lorem.jpg and converting it to grayscale and then contrast stretching it to full dynamic range (gray 0 to gray 255). Then you are creating an orange and white image and appending them. Then you stretch the appending image out to 1024 pixels so that it smears the colors between orange and white. Once you have the color stretched image, it is applied via -clut to colorize the grayscale image so that orange corresponds to full black and white corresponds to full white in your gray image and all the orange-white colors in between correspond to the shades of gray.

**Reply by keyurshah070** [reply]
Thank you so much for detail explanation, much-appricated @fmw42
